File: diagonalDifference.py
def diagonalDifference(arr):
    mainSum = 0
    secondarySum = 0
    for i in range( len( arr ) ):
        mainSum += arr[ i ][ i ]
        secondarySum += arr[ i ][ len( arr ) - i - 1 ]
    diff = mainSum - secondarySum
    return abs( diff )


    
random = [[11, 2, 4], [4, 5, 6], [10, 8, -12]]

print( diagonalDifference ( random ) )

File: diagonalDifference.js
const squareMatrix = [[1,2,3], [1,2,3], [1,2,3]];

function diagonalDifference(arr) {
    // Write your code here
    let length = arr.length - 1;
    let rightDiagonal = 0;
    for ( let i = 0; i < arr.length; i++ ){
        rightDiagonal += arr[i][i];
    }
    let leftDiagonal = 0;
    for ( let i = 0; i < arr.length; i++ ){
        leftDiagonal += arr[i][length];
        length -= 1;
    }
    return Math.abs(rightDiagonal - leftDiagonal)
}

console.log(diagonalDifference(squareMatrix));